integration-tests/NonBlockableReadsTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * Test case for Exceptions.
+ */
+
+declare(strict_types=1);
+
+namespace WebSocket;
+
+use PHPUnit\Framework\TestCase;
+
+class NonBlockingReadsTest extends TestCase
+{
+    private float $start;
+
+    public function setUp(): void
+    {
+        $this->start = hrtime(true);
+    }
+
+    private function onTerm($_): void {
+        // 0.1s
+        $this->assertLessThan(100000, hrtime(true) - $this->start);
```

Comment:
We should avoid assertions in child processes when using PHPUnit.

IMHO, it would make more sense to have `Server` as a child process, and perform assertions in main process, where `Client` would be present.

Reply:
You can always spawn both child and server in a fork, so that you can kill them as you wish, but assertions should be done in the main process.

It might also make more sense to just move child code to a separate .php file, and execute it via `proc_open` ... it should be easier to understand, and it avoid many issues that forking brings.